testing/integration/otel_helm_test.go

```diff
@@ -0,0 +1,189 @@
+
+			settings := cli.New()
+			settings.SetNamespace(testNamespace)
+			actionConfig := &action.Configuration{}
+
+			helmChart, err := loader.Load(chartLocation)
+			require.NoError(t, err, "failed to load helm chart")
+
+			err = actionConfig.Init(settings.RESTClientGetter(), settings.Namespace(), "",
+				func(format string, v ...interface{}) {})
+			require.NoError(t, err, "failed to init helm action config")
+
+			// Initialize a map to hold the parsed data
+			helmValues := make(map[string]any)
+
+			options := values.Options{
+				ValueFiles: []string{tc.valuesFile},
+				Values:     []string{fmt.Sprintf("defaultCRConfig.image.repository=%s", agentImageRepo), fmt.Sprintf("defaultCRConfig.image.tag=%s", agentImageTag)},
+
+				// override secrets reference with env variables
+				JSONValues: []string{
+					fmt.Sprintf(`collectors.cluster.env[1]={"name":"ELASTIC_ENDPOINT","value":"%s"}`, esHost),
+					fmt.Sprintf(`collectors.cluster.env[2]={"name":"ELASTIC_API_KEY","value":"%s"}`, esAPIKey),
+					fmt.Sprintf(`collectors.daemon.env[2]={"name":"ELASTIC_ENDPOINT","value":"%s"}`, esHost),
+					fmt.Sprintf(`collectors.daemon.env[3]={"name":"ELASTIC_API_KEY","value":"%s"}`, esAPIKey),
+				},
+			}
+			providers := getter.All(settings)
+			helmValues, err = options.MergeValues(providers)
+			if err != nil {
+				require.NoError(t, err, "failed to helm values")
+			}
+
+			t.Cleanup(func() {
+				if t.Failed() {
+					dumpLogs(t, ctx, client, testNamespace, testLogsBasePath)
+				}
+
+				uninstallAction := action.NewUninstall(actionConfig)
+				uninstallAction.Wait = true
+
+				_, err = uninstallAction.Run(tc.helmReleaseName)
+				if err != nil {
+					require.NoError(t, err, "failed to uninstall helm chart")
+				}
+			})
+
+			installAction := action.NewInstall(actionConfig)
+			installAction.Namespace = testNamespace
+			installAction.CreateNamespace = true
+			installAction.UseReleaseName = true
+			installAction.ReleaseName = tc.helmReleaseName
+			installAction.Timeout = 2 * time.Minute
+			installAction.Wait = true
+			installAction.WaitForJobs = true
+			_, err = installAction.Run(helmChart, helmValues)
+			require.NoError(t, err, "failed to install helm chart")
+
+			// Pods are created by the OpenTelemetry Operator, it
+			// takes some time for the OpenTelemetry Operator to be
+			// ready
+			require.Eventually(t, func() bool {
+				podList := &corev1.PodList{}
+				err = client.Resources(testNamespace).List(ctx, podList)
+				require.NoError(t, err, fmt.Sprintf("failed to list pods in namespace %s", testNamespace))
+
+				checkedAgentContainers := 0
+
+				for _, pod := range podList.Items {
+					if !strings.HasPrefix(pod.GetName(), tc.helmReleaseName) {
+						continue
+					}
+
+					checkedAgentContainers++
+				}
+				return checkedAgentContainers >= tc.atLeastValidatedPodsNumber
+			}, 5*time.Minute, 10*time.Second, fmt.Sprintf("at least %d agent containers should be checked", tc.atLeastValidatedPodsNumber))
+		})
```

Comment:
qq: can/should we exec inside each pod and do any inside pod checks? In plaint terms if the corrector experience any (early at least) failures will the pod status change accordingly? I am asking because for non-Otel elastic-agent this is not the case 🙂

Reply:
That's a great idea, for now I made the test to assert all the expected pods are in the "running" state. This ensures the "kube-stack" values file had been launch correctly and there are now config/version errors. https://github.com/elastic/elastic-agent/pull/5822/commits/0111a969e3ae89d8986e6c0a308207545a1ca0d2

I think it would be great to have some kind of data ingestion test using the provided Elasticsearch client, that would ensure that is Pod is correctly gathering the expected signals.  What do you think if we add these data assertion tests in a follow-up PR?

Reply:
sure thing we need to add such data ingestion checks, in all k8s tests. I am preparing a mild refactor PR for the k8s tests which will come after this PR (don't want to cause any conflicts here) that hopefully will make it easier to extend existing test to have ingestion checks in the future